ergo/platforms/predictit.py

```diff
@@ -0,0 +1,332 @@
+    >>> q_house_dem = pi.search_market("who control house").search_question("dem")
+    >>> implied = q_pres_dem.get_community_prediction() * q_senate_dem.get_community_prediction() * q_house_dem.get_community_prediction()
+
+    >>> q_sweep = metaculus.get_question(3504)
+    >>> q_sweep.submit(implied)
+    <Response [202]>
+"""
+import re
+from typing import Any, Dict, Iterator, List
+
+import pandas as pd
+import requests
+from dateutil.parser import parse
+from fuzzywuzzy import fuzz
+
+from ergo.distributions.base import flip
+
+
+class PredictItQuestion:
+    """
+    A single binary question in a PredictIt market.
+
+    :param market: PredictIt question instance
+    :param data: Contract JSON retrieved from PredictIt API
+
+    :ivar PredictItQuestion question: predictit question instance
+    :ivar int id: id of the contract
+    :ivar datetime.datetime dateEnd: end-date of a market, usually None
+    :ivar str image: url of the image resource for the contract
+    :ivar str name: name of the contract
+    :ivar str shortName: shortened name of the contract
+    :ivar str status: status of the contract. Closed markets aren't included in the API, so always "Open"
+    :ivar float lastTradePrice: last price the contract was traded at
+    :ivar float bestBuyYesCost: cost to buy a single Yes share
+    :ivar float bestBuyNoCost: cost to buy a single No share
+    :ivar float bestSellYesCost: cost to sell a single Yes share
+    :ivar float bestSellNoCost: cost to sell a single No share
+    :ivar float lastClosePrice: price the contract closed at the previous day
+    :ivar int displayOrder: position of the contract in PredictIt. Defaults to 0 if sorted by lastTradePrice
+    """
+
+    def __init__(self, market: Any, data: Dict):
+        self.market = market
+        self._data = data
+
+    def __repr__(self):
+        if self._data:
+            return f'<PredictItQuestion title="{self.name}">'
+        else:
+            return "<PredictItQuestion>"
+
+    def __str__(self):
+        return repr(self)
+
+    def __getattr__(self, name: str):
+        """
+        If an attribute isn't directly on the class, check whether it's in the
+        raw contract data. If it's a time, format it appropriately.
+        :param name:
+        :return: attribute value
+        """
+        if name in self._data:
+            if name == "dateEnd":
+                if self._data[name] == 'N/A':
+                    return None
+                try:
+                    return parse(self._data[name])
+                except ValueError:
+                    print(
+                        f"The column {name} could not be converted into a datetime"
+                    )
+                    return self._data[name]
+            return self._data[name]
+        else:
+            raise AttributeError(
+                f"Attribute {name} is neither directly on this class nor in the raw question data"
+            )
```

Comment:
If this is the same functionality as in MetaculusQuestion, should factor it out and reuse.

Reply:
It is slightly different, as it has to check to see if the value is 'N/A', and if it is, return a null date. It also uses "dateEnd" instead of "_time".